samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/facade/AssessmentGradingFacadeQueries.java

```diff
@@ -3398,7 +3398,7 @@ public void autoSubmitAssessments() {
 						" where a.publishedAssessmentId = c.assessment.publishedAssessmentId " +
 						" and c.retractDate <= ?" +
 						" and a.status not in (5) and (a.hasAutoSubmissionRun = 0 or a.hasAutoSubmissionRun is null) and c.autoSubmit = 1 " +
-						" and a.submittedDate is not null and a.attemptDate <= c.retractDate " +
+						" and a.submittedDate is not null and (a.attemptDate <= c.retractDate or c.lateHandling = 2) " +
```

Comment:
Yes, you are right. I haven't seen that. I have done some tests and I _think_ that this alternative condition would work fine:

and (a.attemptDate <= c.retractDate or (c.dueDate < now() and c.lateHandling=2))

What do you think, Sam? If lateHandling is 2, retractDate is not used and dueDate is the field to check for autosubmit, isn't it?

The query I am using to test this in MySql while there are pending submissions is:

select a.status, a.hasAutoSubmissionRun, c.autoSubmit, c.lateHandling, assessmentgradingid, publishedassessmentid, submittedDate, attemptDate, startDate, dueDate, retractDate
from sam_AssessmentGrading_t a, sam_publishedAccessControl_t c
where a.publishedAssessmentId = c.assessmentId 
and c.retractDate <= now()
and a.status not in (5) and (a.hasAutoSubmissionRun = 0 or a.hasAutoSubmissionRun is null) and c.autoSubmit = 1 
and a.submittedDate is not null 
and (a.attemptDate <= c.retractDate or (c.dueDate < now() and c.lateHandling=2))
order by a.publishedAssessmentId, a.agentId, a.forGrade desc, a.assessmentGradingId